src/m4/jtr_x86_logic.m4

```diff
@@ -34,6 +34,16 @@ dnl TODO: Ultimately we should not depend on any predefined stuff in arch.h
 dnl at all
 dnl
 AC_DEFUN([JTR_X86_SPECIAL_LOGIC], [
+if test "x$simd" = xno ; then
+if test "x$CC" != xgcc ; then
+echo --disable-simd only tested with gcc for Intel
+AC_SUBST([JOHN_NO_SIMD],[""])
+else
+AC_SUBST([JOHN_NO_SIMD],["-DJOHN_NO_SIMD -mno-sse2 -mno-mmx"])
+CFLAGS="-mno-sse2 -mno-mmx $CFLAGS"
+fi
+fi
```

Comment:
First, you can't just test for `$CC != "gcc"` beause it may be eg. `/usr/bin/gcc` or `/usr/local/bin/gcc-5.2`.

Also, printing a warning is fine but we should definitely still obey the option in case of non-intel. Let the user try - in many cases it may work.

Reply:
BTW on macOS native tools, /usr/bin/gcc is actually a clang so a name test is not enough anyway you do it.

Reply:
It did not work anyway.  Look at .Travis.    The failures for OpenCL gcc builds,fail in configure, saying it can find CL.h but it can not compile it.  Configure knows, but then the script goes ahead and calls for the gpu tests and the john process does not know about them, so the script fails.

Reply:
So that test was a total red herring